### ISSUE TITLE
CQLDataLoader doesn't use keyspace unless it's being created.

### ISSUE THREAD

**Issue by OrangeDog** [jsevellec/cassandra-unit#160]
Means the keyspace name has to be duplicated in the cql script, instead of just setting it in the DataSet.

**Comment by jsevellec** [comment]
Hi,

You can create a CQLDataset by specifying that you let cassandra-unit create a use his own keyspace for the tests by setting keyspaceCreation to true

Is it not what you need?

Jérémy

**Comment by OrangeDog** [comment]
No, the keyspace already exists, so I don't want `keyspaceCreation`.
I just want it to execute `USE keyspace;` consistently, and independently of `keyspaceCreation`.

**Comment by jsevellec** [comment]
ok got it

**Comment by jsevellec** [comment]
Still valid, and I confirmed it against the current code. `CQLDataLoader.initKeyspaceContext` issues `USE` only inside `if (dataSet.isKeyspaceCreation())`, and `load()` issues it again *after* executing the statements. So with `keyspaceCreation` off, your script runs against whatever keyspace the session happened to be on. It usually works only because a previous load left the session pointing at the right place. Documented as a trap for now; the real fix is to `USE` before executing whenever a keyspace name is set.

Keeping this open. (Triage pass after the Cassandra 5.0 / JDK 17 modernization, #345.)
